### PR TITLE
feat: Bitwarden provider (helmfile#276)

### PR DESCRIPTION
**Pull request by smerschjohann**
This PR adds support for the Bitwarden Vaulmanagement API available in the Bitwarden CLI.

### CODE REVIEW

File: pkg/providers/bitwarden/bitwarden.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+	p.Address = cfg.String("address")
+
+	if p.Address == "" {
+		p.Address = os.Getenv("BW_API_ADDR")
+		if p.Address == "" {
+			p.Address = "http://localhost:8087"
+		}
+	}
```
Comment (yxxhero) [inline pkg/providers/bitwarden/bitwarden.go:39]: @smerschjohann thanks so much. please make the `if` logic be more readable.

File: pkg/providers/bitwarden/bitwarden.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	splits := strings.Split(key, "/")
+
+	itemId := splits[0]
+	keyType := "password"
+
+	if len(splits) > 1 {
+		keyType = splits[1]
+	}
+
+	if !(keyType == "username" || keyType == "password" || keyType == "uri" || keyType == "notes") {
```
Comment (yxxhero) [inline pkg/providers/bitwarden/bitwarden.go]: slices.Contains

File: pkg/providers/bitwarden/bitwarden.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+	return p
+}
+
+func (p *provider) GetString(key string) (string, error) {
+	splits := strings.Split(key, "/")
+
+	itemId := splits[0]
+	keyType := "password"
```
Comment (yxxhero) [inline pkg/providers/bitwarden/bitwarden.go]: what will happen if key is empty?

**Comment by yxxhero** [comment]
@smerschjohann please add dome unitests.

**Comment by smerschjohann** [comment]
@yxxhero thx for the review, I integrated your suggestions and also added some Unit Tests.

File: pkg/providers/bitwarden/bitwarden_test.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+		t.Errorf("Expected %s, but got %s", expected, result)
+	}
+
+	// Test case 3: cfgAddress and envAddr are empty
+	os.Setenv("BW_API_ADDR", "")
+	expected = "http://localhost:8087"
+	result = getAddressConfig("")
+	if result != expected {
+		t.Errorf("Expected %s, but got %s", expected, result)
+	}
```
Comment (yxxhero) [inline pkg/providers/bitwarden/bitwarden_test.go:33]: unset env

Comment (yxxhero) [reply pkg/providers/bitwarden/bitwarden_test.go:33]: @smerschjohann then I will merge this. thanks so much.

Comment (smerschjohann) [reply pkg/providers/bitwarden/bitwarden_test.go:33]: @yxxhero done

**Review by yxxhero — APPROVED** [review]

**Pull request merged**